src/main/sensors/gyro.c

```diff
@@ -935,13 +929,19 @@ static FAST_CODE void gyroUpdateSensor(gyroSensor_t *gyroSensor, timeUs_t curren
             gyroADCf = gyroSensor->fastKalmanApplyFn((filter_t *)&gyroSensor->fastKalman[axis], gyroADCf);
 #elif defined(USE_GYRO_BIQUAD_RC_FIR2)
             gyroADCf = gyroSensor->biquadRCFIR2ApplyFn((filter_t *)&gyroSensor->biquadRCFIR2[axis], gyroADCf);
-#endif
-#ifdef USE_GYRO_DATA_ANALYSE
-            gyroADCf = gyroSensor->notchFilterDynApplyFn((filter_t *)&gyroSensor->notchFilterDyn[axis], gyroADCf);
 #endif
             gyroADCf = gyroSensor->notchFilter1ApplyFn((filter_t *)&gyroSensor->notchFilter1[axis], gyroADCf);
             gyroADCf = gyroSensor->notchFilter2ApplyFn((filter_t *)&gyroSensor->notchFilter2[axis], gyroADCf);
             gyroADCf = gyroSensor->softLpfFilterApplyFn(gyroSensor->softLpfFilterPtr[axis], gyroADCf);
+
+#ifdef USE_GYRO_DATA_ANALYSE
+            if (isDynamicFilterActive()) {
```

Comment:
Given this `push` is just one in-place addition, maybe it's better to just omit the `if` and push data unconditionally.

Reply:
I think this is the `FEATURE` being turned on, so probably good.

Reply:
Yes, but memory is still there, and just adding to it even if it never gets used is cheaper than checking for feature and then adding.